sessionize/app/src/main/java/co/touchlab/sessionize/event/EventFragment.kt

```diff
@@ -91,7 +91,9 @@ class EventFragment : Fragment() {
             fab.visibility = View.GONE
         } else {
             fab.setOnClickListener {
+
                 eventViewModel.eventModel.toggleRsvp(!event.isRsvped())
+                eventViewModel.eventModel.handleLocalNotification(event.session)
```

Comment:
I think it makes sense to move this into toggleRsvp(). My philosophy here is that the communication from the view to the viewModel should be to notify the viewModel about user actions, not directing what should happen due to that action. So in this case the view can just say "the user toggled rsvp" and the decision of "what things should happen when the user rsvp's" is in the (shared) business logic layers.

Reply:
That's true, that would make more sense. The reason I had them separate was because the EventModel doesn't hold a reference to the session, only to the sessionId. I could either pass the entire event into the toggleRSVP or have the EventModel have a reference to the event or something similar. Or I could lookup the session info based on the id, but that seems repetitive.

Reply:
I don't have a problem with you passing in the session/event since it's not any different/worse than passing it on the next line. It really seems like the model could be refactored to allow more of the event data to be available (or we need to define a strategy for how we want things like this to be handled), but we do have another story for that.

Reply:
Moved notification call to toggleRsvp